crates/storage/store/storage.rs

```diff
@@ -210,51 +218,87 @@ impl Store {
     }
 
     /// Add transaction to the pool
-    pub fn add_transaction_to_pool(
-        &self,
-        hash: H256,
-        transaction: MempoolTransaction,
-    ) -> Result<(), StoreError> {
-        self.engine.add_transaction_to_pool(hash, transaction)
-    }
-
-    /// Get a transaction from the pool
-    pub fn get_transaction_from_pool(
-        &self,
-        hash: H256,
-    ) -> Result<Option<MempoolTransaction>, StoreError> {
-        self.engine.get_transaction_from_pool(hash)
+    pub fn add_transaction_to_pool(&self, transaction: MempoolTransaction) {
+        let mut mempool = self.mempool.lock().unwrap();
+        let maybe_old_value = mempool.get_mut(&transaction.sender());
+        let sender = transaction.sender();
+
+        match maybe_old_value {
+            Some(old_value) => {
+                old_value.insert(transaction.nonce(), transaction);
+            }
+            None => {
+                let mut new_value = BTreeMap::new();
+                new_value.insert(transaction.nonce(), transaction);
+                mempool.insert(sender, new_value);
+            }
+        }
     }
 
     /// Add a blobs bundle to the pool by its blob transaction hash
-    pub fn add_blobs_bundle_to_pool(
-        &self,
-        tx_hash: H256,
-        blobs_bundle: BlobsBundle,
-    ) -> Result<(), StoreError> {
-        self.engine.add_blobs_bundle_to_pool(tx_hash, blobs_bundle)
+    pub fn add_blobs_bundle_to_pool(&self, tx_hash: H256, blobs_bundle: BlobsBundle) {
+        self.blobs_bundle_pool
+            .lock()
+            .unwrap()
+            .insert(tx_hash, blobs_bundle);
     }
 
     /// Get a blobs bundle to the pool given its blob transaction hash
-    pub fn get_blobs_bundle_from_pool(
-        &self,
-        tx_hash: H256,
-    ) -> Result<Option<BlobsBundle>, StoreError> {
-        self.engine.get_blobs_bundle_from_pool(tx_hash)
+    pub fn get_blobs_bundle_from_pool(&self, tx_hash: H256) -> Option<BlobsBundle> {
+        self.blobs_bundle_pool
+            .lock()
+            .unwrap()
+            .get(&tx_hash)
+            .cloned()
     }
 
     /// Remove a transaction from the pool
-    pub fn remove_transaction_from_pool(&self, hash: H256) -> Result<(), StoreError> {
-        self.engine.remove_transaction_from_pool(hash)
+    pub fn remove_transaction_from_pool(&self, address: Address, nonce: u64) {
+        if let Some(old_value) = self.mempool.lock().unwrap().get_mut(&address) {
+            let tx = old_value.get(&nonce).unwrap();
```

Comment:
Do we know for sure that the transaction exists here? Maybe we can obtain it from the `remove` method

Reply:
Is this what you meant [730aa10](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1002/commits/730aa101663d7d0150a8cd7696ead00f67e905cc)?

Reply:
Yes!